src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java

```diff
@@ -218,26 +237,14 @@ public void onStarted(Run run, TaskListener listener) {
                 DatadogEvent event = new BuildStartedEventImpl(buildData);
                 client.event(event);
             }
-            // Send a metric
-            // item.getInQueueSince() may raise a NPE if a worker node is spinning up to run the job.
-            // This could be expected behavior with ec2 spot instances/ecs containers, meaning no waiting
-            // queue times if the plugin is spinning up an instance/container for one/first job.
-            Queue queue = getQueue();
-            Queue.Item item = queue.getItem(run.getQueueId());
+
             Map<String, Set<String>> tags = buildData.getTags();
             String hostname = buildData.getHostname(DatadogUtilities.getHostname(null));
-            try (Metrics metrics = client.metrics()) {
-                long waitingMs = (DatadogUtilities.currentTimeMillis() - item.getInQueueSince());
-                metrics.gauge("jenkins.job.waiting", TimeUnit.MILLISECONDS.toSeconds(waitingMs), hostname, tags);
 
-                PipelineQueueInfoAction queueInfoAction = run.getAction(PipelineQueueInfoAction.class);
-                if (queueInfoAction != null) {
-                    queueInfoAction.setQueueTimeMillis(waitingMs);
+            if (waitingMs != null) {
+                try (Metrics metrics = client.metrics()) {
+                    metrics.gauge("jenkins.job.waiting", TimeUnit.MILLISECONDS.toSeconds(waitingMs), hostname, tags);
```

Comment:
Not sure if this metric has the same meaning than CI Visibility Queue Time, but here we're not taking into account the "propagated" queue time (for Jenkins Declarative Pipelines using `agent`)

Reply:
I'm not sure either, the documentation just says "Time spent waiting for job to run".
I'd leave it as is, at least for now, given that there haven't been any customer complaints